hpat/tests/test_series.py

```diff
@@ -1685,35 +1740,26 @@ def test_impl(S):
         S = pd.Series([73, 21, 10005, 5, 1], index=[2, 3, 4, 5, 6])
         np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
 
-    def test_series_nsmallest1(self):
-        def test_impl(S):
-            return S.nsmallest(4)
-        hpat_func = hpat.jit(test_impl)
-
-        m = 100
-        np.random.seed(0)
-        S = pd.Series(np.random.randint(-30, 30, m))
-        np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
-
-    def test_series_nsmallest_default1(self):
-        def test_impl(S):
-            return S.nsmallest()
+    def test_series_nsmallest(self):
+        def test_impl():
+            series = pd.Series([1.0, np.nan, -1.0, 0.0, 5e-324])
+            return series.nsmallest(4)
         hpat_func = hpat.jit(test_impl)
 
-        m = 100
-        np.random.seed(0)
-        S = pd.Series(np.random.randint(-30, 30, m))
-        np.testing.assert_array_equal(hpat_func(S).values, test_impl(S).values)
+        np.testing.assert_array_equal(hpat_func(), test_impl())
```

Comment:
[Pandas docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.nsmallest.html): nsmallest returns Series.
Consider use pandas.testing.assert_series_equal(). It knows how to compare series better.
```suggestion
        pandas.testing.assert_series_equal(hpat_func(), test_impl())
```

Reply:
@PokhodenkoSA You are right, but functionality "in old style" doesn't work with indexes correctly. So until we completely moved to "new style" the check is valid. Now we don't drop code "in old style" to keep MPI parallelism.

Reply:
@densmirn A better solution would be to change the test as @PokhodenkoSA suggested and skip it if executed with old-style (as there's a bug which has to be fixed anyway), while new-style impl can be written so that it handles index correctly from the start.

Reply:
If I skipped the tests it would be no tests "in old style" at all. Or do you propose to duplicate the tests for "old style" and "new style" code with different asserts?

Reply:
@densmirn please use
```
if hpat.config.config_pipeline_hpat_default:
  np.testing.assert_array_equal(hpat_func(), test_impl())
else
  pd.testing.assert_series_equal(hpat_func(), test_impl())

```